packages/passwordless-auth-construct/src/custom-auth/custom_auth_service.ts

```diff
@@ -280,31 +325,57 @@ export class CustomAuthService {
    * @param deliveryMedium - The delivery medium for the challenge.
    * @param event - The Create Auth Challenge event.
    * @returns An object that contains a boolean indicating if the destination is
-   * verified, and the destination.
+   * verified, and the destination and if is the first attempt for passwordless signUp
    */
   private validateDestination = (
     deliveryMedium: DeliveryMedium,
     event: CreateAuthChallengeTriggerEvent
-  ):
-    | { isVerified: true; destination: string }
-    | { isVerified: false; destination: undefined } => {
+  ): {
+    isFirstSignInAttempt: boolean;
+    isVerified: boolean;
+    destination: string;
+  } => {
     const {
       email,
       phone_number: phoneNumber,
       email_verified: emailVerified,
       phone_number_verified: phoneNumberVerified,
+      [PASSWORDLESS_SIGN_UP_ATTR_NAME]: passwordlessSignUp,
     } = event.request.userAttributes;
+
+    let isFirstSignInAttempt = false;
+    try {
+      const passwordlessConfiguration = JSON.parse(passwordlessSignUp);
+      if (
+        passwordlessConfiguration.allowSignInAttempt &&
+        passwordlessConfiguration.deliveryMedium === deliveryMedium
+      ) {
+        isFirstSignInAttempt = true;
+      }
+    } catch (_err) {
+      // best effort to parse passwordless_sign_up attribute if the user was created via passwordless sign up
```

Comment:
question: How are you planning on handling when the value can not be parsed? Might be good to return an error to to client.

Reply:
after re-reading this snippet, looks like a no-op is sufficient.